### PR TITLE
Return targets with spectra by observatory

### PR DESCRIPTION
**Pull request by imedan**
This adds the feature to return the list of targets with spectra at some observatory. Currently the query works by specifying a data release, an observatory and an instrument.

For just searching on BOSS or APOGEE spectra, the queries seem optimized enough I think (~4.5 sec for BOSS and ~2 sec for APOGEE at APO). Doing the union of both instruments takes much longer (~45 sec). The bottleneck here seems to be sequential scanning from looking at the query plan. I tried to turn this off and test it again, but the query plan says it still uses sequential scanning even after doing this. So, I am unsure what do do about this bit of the feature.

### CODE REVIEW

**Comment by havok2063** [comment]
@imedan Is this ready to be reviewed?  If so, can you mark it as ready.

File: python/valis/db/queries.py [hunk 1]
```diff
@@ -264,6 +264,62 @@ def carton_program_search(name: str, name_type: str) -> peewee.ModelSelect:
...
+    ----------
+    release : str
+        the data release to look up
+
+    obs: str
+        Observatory to get targets from. Either 'APO' or 'LCO'
+
+    obsWave: str
+        Which spectrograph to return data from. Can be 'boss',
+        'apogee' or 'all' for both.
```
Comment (havok2063) [inline python/valis/db/queries.py:280]: should this parameter be renamed `spectrograph` or `instrument` to make it more explicit?

File: python/valis/routes/query.py [hunk 2]
```diff
@@ -147,3 +148,19 @@ async def carton_program(self,
...
+    @router.get('/obs', summary='Return targets with spectrum at observatory',
+                response_model=List[SDSSidStackedBase], dependencies=[Depends(get_pw_db)])
+    async def obs(self,
+                  release: Annotated[str, Query(description='Data release to query', example='IPL3')],
+                  obs: Annotated[str,
+                                 Query(enum=['APO', 'LCO'],
+                                       description='Observatory to get targets from. Either "APO" or "LCO"',
+                                       example='APO')] = 'APO',
+                  obsWave: Annotated[str,
+                                     Query(enum=['boss', 'apogee', 'all'],
```
Comment (havok2063) [inline python/valis/routes/query.py:160]: Can you specify default values for `obs` and `obsWave`?

**Review by havok2063 — COMMENTED** [review]
This looks good.  The timings might be ok.  I think users will only call this endpoint when they explicitly want an observatory.  By default, queries select everything anyways, so if they don't choose an option, this query doesn't need to be run.

File: python/valis/db/queries.py (hunk 1 above)
Comment (imedan) [reply python/valis/db/queries.py]: This makes sense. I have switched this to `spectrograph`.

File: python/valis/routes/query.py (hunk 2 above)
Comment (imedan) [reply python/valis/routes/query.py]: Are default values not set with how it is already written? I assumed having the `Query(...) = ...` set the default value.

Comment (havok2063) [reply python/valis/routes/query.py]: Yeah you are right!

**Pull request merged**